roles/schema_registry/tasks/main.yml

```diff
@@ -142,9 +142,19 @@
   tags:
     - configuration
 
-- name: Configure RBAC
+- name: Configure RBAC for OAuth User
   include_tasks: rbac.yml
-  when: rbac_enabled|bool
+  vars:
+    sr_user: "{{schema_registry_oauth_user}}"
+  when: rbac_enabled|bool and oauth_enabled
+
+- name: Configure RBAC for LDAP User
+  include_tasks: rbac.yml
+  vars:
+    sr_user: "{{schema_registry_ldap_user}}"
+  when:
+    - rbac_enabled|bool
+    - not oauth_enabled or ( oauth_enabled and ldap_with_oauth_enabled )
```

Comment:
1. maybe we can hav parenthesis here so that it becomes clear without knowing the operator precendence of not and or.
2. Just curious in case of both ldap and oauth combined mode why are we using ldap_user and not oauth_user

Reply:
1. makes sense, it will also keep us safe in case something changes in future ansible versions
2. the task previous to this is already taking care of oauth user, in ldap_with_oauth it is necessary to have oauth_enabled, which satisfies the condition of the previous task

Reply:
2. In task before this we have condn rbac+oauth there we use oauth user. In this task we have condn either (rbac+ldap) or we have (rbac+ldap+oauth) then we are using ldap user. My idea was that maybe we can seperate this task into 2 further tasks and in the end have 3 tasks ->
1. rbac + ldap  ->  ldap_user
2. rbac+ oauth -> oauth_user
3. rbac+ oauth + ldap -> oauth_user 
Reason for having oauth_user in case of combined mode is that later when someone wants to move to oauth only from oauth + ldap then at that time they wont have to change ldap user to oauth user

Reply:
didn't get it completely, but ldap user and oauth user would always be different, one is on ldap sever and another is on idp

Reply:
I think the understanding here is wrong, when we have rbac+ldap, we need the ldap_user to have the required permissions. If we use your conditions, ldap_user will not work in case of ldap_with_oauth since you are giving permissions to the oauth_user twice instead of ldap_user

Reply:
ohh my bad. got it now.